// firebase-tools/lib/mcp/tools/core/consult_assistant.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.consult_assistant = void 0;
const zod_1 = require("zod");
const tool_1 = require("../../tool");
const util_1 = require("../../util");
const fdcExperience_1 = require("../../../gemini/fdcExperience");
exports.consult_assistant = (0, tool_1.tool)({
    name: "consult_assistant",
    description: "Send a question to an AI assistant specifically enhanced to answer Firebase questions.",
    inputSchema: zod_1.z.object({
        prompt: zod_1.z
            .string()
            .describe("A description of what the user is trying to do or learn with Firebase."),
    }),
    annotations: {
        title: "Consult Firebase Assistant",
        readOnlyHint: true,
    },
    _meta: {
        requiresProject: true,
        requiresAuth: true,
        requiresGemini: true,
    },
}, async ({ prompt }, { projectId }) => {
    const schema = await (0, fdcExperience_1.chatWithFirebase)(prompt, projectId);
    return (0, util_1.toContent)(schema);
});
